modules/references/client/components/CreateReference.component.js

```diff
@@ -35,14 +35,16 @@ export default function CreateReference({ userFrom, userTo }) {
 
   useEffect(() => {
     (async () => {
-      const reference = await api.references.read({
+      const references = await api.references.read({
         userFrom: userFrom._id,
         userTo: userTo._id,
       });
 
-      if (reference.length === 1) {
-        setIsDuplicate(true);
-      }
+      references.forEach(reference => {
+        if (reference.userFrom.username === userFrom.username) {
+          setIsDuplicate(true);
```

Comment:
Should we short-circuit the `forEach` loop here to avoid looping through the rest?